### DISCUSSION TITLE
Add filters to Discussions GraphQL API

### DISCUSSION THREAD

**Discussion by benjavicente** [community/community#111272] [Discussions]
### Select Topic Area

Product Feedback

### Body

The answered filter has been added as an argument to discussions, but most of the filters that are available in the web UI are missing. [Here is the full list of filters](https://docs.github.com/en/search-github/searching-on-github/searching-discussions).

Notably, **there doesn't seem to be a way to filter by label, author, answered by, and involves** by an API call.

Those filters can be used with the pagination `totalCount` to get useful statistics about users and labels. The current workaround requires to manually search thought most discussions to get the desired count.  Issues has `IssueFilters` to do the filtering, a similar API would be great.

---

Similar issue: https://github.com/orgs/community/discussions/9096

**Comment by Unmesh100** [comment]
You're absolutely right. Having comprehensive filters available via API, similar to what's provided in the web UI, would be incredibly beneficial. The ability to filter by label, author, answered by, and involvement would save a ton of manual effort and improve the accuracy of statistics gathering.

This would indeed be a great feature for the API to adopt. As you've mentioned, the IssueFilters in the Issues API are a perfect example of how effective such functionality can be. Implementing similar filters for discussions would streamline a lot of processes.

Have you had a chance to submit this feedback directly to the platform's development team or product feedback channels? They might be very receptive to such a valuable suggestion.
